Fix non-POSIX use of echo -n, don't enable maintainer mode, and don't
look for old automake versions.
--- autogen.sh.orig	2009-09-29 04:02:09.000000000 +1000
+++ autogen.sh	2021-07-21 12:57:57.000000000 +1000
@@ -29,12 +29,12 @@
 # do we need automake?
 if test -r Makefile.am; then
   AM_OPTIONS=`fgrep AUTOMAKE_OPTIONS Makefile.am`
-  AM_NEEDED=`echo $AM_OPTIONS | $VERSIONGREP`
+  AM_NEEDED=""
   if test x"$AM_NEEDED" = "x$AM_OPTIONS"; then
     AM_NEEDED=""
   fi
   if test -z $AM_NEEDED; then
-    echo -n "checking for automake... "
+    printf "checking for automake... "
     AUTOMAKE=automake
     ACLOCAL=aclocal
     if ($AUTOMAKE --version < /dev/null > /dev/null 2>&1); then
@@ -44,7 +44,7 @@
       AUTOMAKE=
     fi
   else
-    echo -n "checking for automake $AM_NEEDED or later... "
+    printf "checking for automake %s or later... " "$AM_NEEDED"
     majneeded=`echo $AM_NEEDED | $VERSIONMKMAJ`
     minneeded=`echo $AM_NEEDED | $VERSIONMKMIN`
     for am in automake-$AM_NEEDED automake$AM_NEEDED \
@@ -60,7 +60,7 @@
       fi
     done
     test -z $AUTOMAKE &&  echo "no"
-    echo -n "checking for aclocal $AM_NEEDED or later... "
+    printf "checking for aclocal %s or later... " "$AM_NEEDED"
     for ac in aclocal-$AM_NEEDED aclocal$AM_NEEDED \
 	aclocal aclocal-1.7 aclocal-1.8 aclocal-1.9 aclocal-1.10; do
       ($ac --version < /dev/null > /dev/null 2>&1) || continue
@@ -84,7 +84,7 @@
   }
 fi
 
-echo -n "checking for libtool... "
+printf "checking for libtool... "
 for LIBTOOLIZE in libtoolize glibtoolize nope; do
   ($LIBTOOLIZE --version) < /dev/null > /dev/null 2>&1 && break
 done
@@ -126,4 +126,4 @@
 autoconf || exit 1
 
 cd $olddir
-$srcdir/configure --enable-maintainer-mode "$@" && echo
+$srcdir/configure "$@"
